From 347b81cecdff33a71509c96c69f58b6e28420229 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Fri, 23 Jun 2017 16:59:40 -0700
Subject: [PATCH] Eleven: Fix missing audio record permission prompt for audio
 visualizer

The audio visualizer is enabled by default and requires the audio record
permission, so ask for it on init.

Change-Id: I321888acdebaeb09af1294cc6b159111d22863e7
---
 src/com/cyanogenmod/eleven/ui/activities/HomeActivity.java | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/com/cyanogenmod/eleven/ui/activities/HomeActivity.java b/src/com/cyanogenmod/eleven/ui/activities/HomeActivity.java
index 2e136eb..35ca10b 100644
--- a/src/com/cyanogenmod/eleven/ui/activities/HomeActivity.java
+++ b/src/com/cyanogenmod/eleven/ui/activities/HomeActivity.java
@@ -72,6 +72,7 @@ public class HomeActivity extends SlidingPanelActivity implements
     public static final int EQUALIZER = 2;
 
     private static final int PERMISSION_REQUEST_STORAGE = 1;
+    private static final int PERMISSION_REQUEST_RECORD_AUDIO = 2;
     private Bundle mSavedInstanceState;
 
     private String mKey;
@@ -97,6 +98,14 @@ public class HomeActivity extends SlidingPanelActivity implements
     }
 
     private void init() {
+        // Get permission for audio visualizer
+        if (checkSelfPermission(Manifest.permission.RECORD_AUDIO) !=
+                    PackageManager.PERMISSION_GRANTED) {
+                requestPermissions(
+                        new String[] {Manifest.permission.RECORD_AUDIO},
+                        PERMISSION_REQUEST_RECORD_AUDIO);
+        }
+
         // if we've been launched by an intent, parse it
         Intent launchIntent = getIntent();
         boolean intentHandled = false;
-- 
2.11.0

